crates/types/src/schema/metadata/updater/mod.rs

```diff
@@ -354,29 +355,39 @@ impl SchemaUpdater {
             .collect::<Result<HashMap<_, _>, _>>()?;
 
         // Did we find an existing deployment with a conflicting endpoint url?
-        let mut existing_deployments = self.schema.deployments.iter().filter(|(_, schemas)| {
-            schemas.ty.protocol_type() == deployment_metadata.ty.protocol_type()
-                && schemas.ty.normalized_address() == deployment_metadata.ty.normalized_address()
-        });
+        let existing_deployment = self
+            .schema
+            .deployments
+            .iter()
+            .filter(|(_, schemas)| {
+                schemas.ty.protocol_type() == deployment_metadata.ty.protocol_type()
+                    && schemas.ty.normalized_address()
+                        == deployment_metadata.ty.normalized_address()
+                    && routing_header.as_ref().is_none_or(
+                        |(routing_header_key, routing_header_value)| {
+                            schemas
+                                .delivery_options
+                                .additional_headers
+                                .get(routing_header_key)
+                                .is_some_and(|v| v == routing_header_value)
+                        },
+                    )
+            })
+            // There are few situations where we might have multiple deployments for the same endpoint:
+            // * If the user specified in the at least two previous registrations the routing-header,
+            //   but then it didn't specify it in this one. In this case, multiple deployments will match the above filter
+            // * If update_deployment was used on at least one deployment, pointing to the same address of another deployment,
+            //   resulting in having two deployments pointing at the same address.
+            //
+            // We pick max_by created_at, because with force the user wants to override the last deployment version, and not old ones.
+            .max_by(|(_, x), (_, y)| x.created_at.cmp(&y.created_at));
```

Comment:
Is there some agreement that all deployments with the same address (but different routing header) are different revisions of the same service? 

I might be missing something but I am under the impression that different routing headers means possibly completely different deployments and services that can be hosted on those deployments. Hence assuming the last deployment added might not be correct. To update a deployment this has to match exactly the routing header. No ?